stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -496,8 +496,10 @@ function _artifact_str(__module__, artifacts_toml, name, path_tail, artifact_dic
 
     # If not, we need to download it.  We look up the Pkg module through `Base.loaded_modules()`
     # then invoke `ensure_artifact_installed()`:
-    Pkg = first(filter(p-> p[1].name == "Pkg", Base.loaded_modules))[2]
-    return jointail(Pkg.Artifacts.ensure_artifact_installed(string(name), artifacts_toml; platform), path_tail)
+    if ensure_artifact_installed === nothing
+        error("Artifact $(repr(name)) was not installed correctly. Try `using Pkg; Pkg.build(\"$__module__\"); Pkg.Artifacts.ensure_all_artifacts_installed($(repr(artifacts_toml)))`?")
```

Comment:
```suggestion
    meta = artifact_meta(name, artifact_dict, artifacts_toml; platform)
    if meta["lazy"]
        error("Artifact $(repr(name)) is a lazy artifact; package developers must `import Pkg` before using lazy artifacts.")
    else
        error("Artifact $(repr(name)) was not installed correctly. Try `using Pkg; Pkg.instantiate()`")
    end
```

There are two cases where this can go wrong; the first is something terrible happened during `Pkg.add()` and we're in an inconsistent state.  As far as I know, this has never happened, but in the event that it does, `Pkg.build()` isn't the right fix, `Pkg.instantiate()` is, I think.

The second is that this is a package developer adding an artifact to their package and being surprised when lazy artifacts don't work.

We can disambiguate the two by looking to see if the corresponding entry in `artifact_dict` marks the artifact as lazy.